File: apps/web/src/application/usecases/getGrammarDetail.ts
import type { GrammarRepository } from "../ports/GrammarRepository";
import type { GrammarEntry } from "../../domain/grammar/entities";

export function getGrammarDetail(
  repository: GrammarRepository,
  grammarId: string,
): Promise<GrammarEntry | null> {
  return repository.getById(grammarId);
}

File: apps/api/app/application/usecases/grammar/get_grammar_detail.py
from app.application.ports.grammar_repository import GrammarRepository
from app.domain.entities.grammar import GrammarEntry


def get_grammar_detail(
    repository: GrammarRepository,
    grammar_id: str,
) -> GrammarEntry | None:
    return repository.get_by_id(grammar_id)
